docs/sql-ref-datetime-pattern.md

```diff
@@ -0,0 +1,212 @@
+  <td> 0 </td>
+</tr>
+<tr>
+  <td> <b>H</b> </td>
+  <td> hour-of-day (0-23)  </td>
+  <td> number </td>
+  <td> 0 </td>
+</tr>
+<tr>
+  <td> <b>m</b> </td>
+  <td> minute-of-hour </td>
+  <td> number </td>
+  <td> 30 </td>
+</tr>
+<tr>
+  <td> <b>s</b> </td>
+  <td> second-of-minute </td>
+  <td> number </td>
+  <td> 55 </td>
+</tr>
+<tr>
+  <td> <b>S</b> </td>
+  <td> fraction-of-second </td>
+  <td> fraction </td>
+  <td> 978 </td>
+</tr>
+<tr>
+  <td> <b>z</b> </td>
+  <td> time-zone name </td>
+  <td> zone-name </td>
+  <td> Pacific Standard Time; PST </td>
+</tr>
+<tr>
+  <td> <b>O</b> </td>
+  <td> localized zone-offset </td>
+  <td> offset-O </td>
+  <td> GMT+8; GMT+08:00; UTC-08:00; </td>
+</tr>
+<tr>
+  <td> <b>X</b> </td>
+  <td> zone-offset 'Z' for zero </td>
+  <td> offset-X </td>
+  <td> Z; -08; -0830; -08:30; -083015; -08:30:15; </td>
+</tr>
+<tr>
+  <td> <b>x</b> </td>
+  <td> zone-offset </td>
+  <td> offset-x </td>
+  <td> +0000; -08; -0830; -08:30; -083015; -08:30:15; </td>
+</tr>
+<tr>
+  <td> <b>Z</b> </td>
+  <td> zone-offset </td>
+  <td> offset-Z </td>
+  <td> +0000; -0800; -08:00; </td>
+</tr>
+<tr>
+  <td> <b>'</b> </td>
+  <td> escape for text </td>
+  <td> delimiter </td>
+  <td></td>
+</tr>
+<tr>
+  <td> <b>''</b> </td>
+  <td> single quote </td>
+  <td> literal </td>
+  <td> ' </td>
+</tr>
+</table>
+
+The count of pattern letters determines the format.
+
+- Text: The text style is determined based on the number of pattern letters used. Less than 4 pattern letters will use the short form. Exactly 4 pattern letters will use the full form.
+
+- Number: If the count of letters is one, then the value is output using the minimum number of digits and without padding. Otherwise, the count of digits is used as the width of the output field, with the value zero-padded as necessary. The following pattern letters have constraints on the count of letters. Only one letter 'F' can be specified. Up to two letters of 'd', 'H', 'h', 'K', 'k', 'm', and 's' can be specified. Up to three letters of 'D' can be specified.
+
+- Number/Text: If the count of pattern letters is 3 or greater, use the Text rules above. Otherwise use the Number rules above.
+
+- Fraction: Outputs the micro-of-second field as a fraction-of-second. The micro-of-second value has six digits, thus the count of pattern letters is from 1 to 6. If it is less than 6, then the micro-of-second value is truncated, with only the most significant digits being output.
+
+- Year: The count of letters determines the minimum field width below which padding is used. If the count of letters is two, then a reduced two digit form is used. For printing, this outputs the rightmost two digits. For parsing, this will parse using the base value of 2000, resulting in a year within the range 2000 to 2099 inclusive. If the count of letters is less than four (but not two), then the sign is only output for negative years. Otherwise, the sign is output if the pad width is exceeded.
```

Comment:
> Otherwise, the sign is output if the pad width is exceeded.

This is not true when G is present, right?

Reply:
Right, emphasize in e846fbb.